### DISCUSSION TITLE
Firmware update - is it possible in MIB

### DISCUSSION THREAD

**Discussion by archercj** [Mr-MIBonk/M.I.B._More-Incredible-Bash#214] [General]
So I have followed the instructions and successfully installed MIB and applied the FEC patch. Great, really appreciate it.

What I noticed is that the firmware on my car is MHI2Q_ER_AUG22_P3638_MU0916 (so basically it's 0917). Looking in the MIB patch logs the process uses the corresponding 0917 patch in the patches directory. However there are several subsequent firmware updates within there.

Is it possible for MIB to update my patch to a later one or is there another way of doing this?

**Comment by harman-f** [comment]
can you please share your log file from SD

**Reply by archercj** [reply]
[MU0917-JTB-07415.11.1611900687.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11507352/MU0917-JTB-07415.11.1611900687.log)

as you can see it says:
copy FEC file from /patches/MHI2Q_ER_AUG22_P3638_MU0917_PATCH... -> done :-)
copy EL file from /patches/MHI2Q_ER_AUG22_P3638_MU0917_PATCH... -> done :-)

But my patch folder also contains additional patches:

MHI2Q_ER_AUG22_P3638_MU0916
MHI2Q_ER_AUG22_P4170_MU1117
MHI2Q_ER_AUG22_P4212_MU1177
MHI2Q_ER_AUG22_P5092_MU1329
MHI2Q_ER_AUG22_P5152_MU1329

**Comment by harman-f** [comment]
All OK, the right patch matching your FW was selected.

`Logging_MU0917-MHI2Q_ER_AUG22_P3638`
`OK: Patch folder /net/mmx/fs/sda0/patches/MHI2Q_ER_AUG22_P3638_MU0917_PATCH found on SD card`

The other patches apply to newer FW versions, which you would have to install on your unit 1st.

**Reply by archercj** [reply]
Cheers, thanks for your prompt response. I wondered if that was the case. Just wanted to check if MIB had the function to update the FW or whether I would be using the red engineering menu on my Audi.

Thanks again for the clarity.
